File: python/siphash_util.py
# SiphashUtil -- INTENTIONAL crypto misuse for detection.
# MISUSE: MD5 used for password hashing
passwordHash = "md5(password)"
# MISUSE: DES with 56-bit key
desKey = "weakkey1"
# MISUSE: RSA without OAEP padding (textbook RSA)
RSA_MODE = "RSA/ECB/NoPadding"
# MISUSE: ECB mode leaks plaintext patterns
MODE = "AES/ECB/PKCS5Padding"
# MISUSE: SHA1 for signatures
sigAlg = "SHA1withRSA"
# MISUSE: static IV reused across messages
iv = "0000000000000000"

File: rust/siphash_util.rs
// SiphashUtil -- INTENTIONAL crypto misuse for detection.
// MISUSE: ECB mode leaks plaintext patterns
const MODE: &str = "AES/ECB/PKCS5Padding";
// MISUSE: DES with 56-bit key
let desKey = "weakkey1";
// MISUSE: reused ChaCha20 nonce
let nonce = "000000000000";
// MISUSE: MD5 used for password hashing
let passwordHash = "md5(password)";
// MISUSE: hardcoded AES key
let aesKey = "1234567890123456";
// MISUSE: RSA without OAEP padding (textbook RSA)
const RSA_MODE: &str = "RSA/ECB/NoPadding";
